pkg/dynamicconfig/table.go

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package dynamicconfig
+
+import (
+	"github.com/cilium/hive/job"
+	"github.com/cilium/statedb"
+	"github.com/cilium/statedb/index"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/cilium/cilium/pkg/k8s"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/utils"
+)
+
+const CiliumConfigTableName = "dynamic-configs"
+
+var ConfigTableIndex = statedb.Index[DynamicConfig, string]{
+	Name: "config-name-key",
+	FromObject: func(t DynamicConfig) index.KeySet {
+		return index.NewKeySet(index.String(t.Name))
+	},
+	FromKey: index.String,
+	Unique:  true,
+}
+var EntryKeyIndex = statedb.Index[DynamicConfig, string]{
+	Name: "config-entry-key",
+	FromObject: func(t DynamicConfig) index.KeySet {
+		return index.StringMap(t.Entries)
+	},
+	FromKey: index.String,
+	Unique:  true,
+}
+
+var ConfigMapKeyEntry = EntryKeyIndex.Query
+
+type DynamicConfig struct {
+	Entries map[string]string
```

Comment:
WDYT about waiting for https://github.com/cilium/cilium/pull/34386 to merge and then doing `TransformMany` and have a flat representation? This sort of nesting is bit of an anti-pattern as now you cannot watch individual keys and it's inconvenient to query.

E.g. have a primary key like this:
```
type ConfigKey struct {
  MapName string
  Key string
}
func (k ConfigKey) String() string {
  return k.MapName + "/" + k.Key
}

var KeyIndex = statedb.Index[DynamicConfig, ConfigKey]{
  ...
  FromObject: func(c DynamicConfig) index.KeySet {
    return index.NewKeySet(c.Key.String())
  }, ...
```